### DISCUSSION TITLE
Stereokit UWP template running errors

### DISCUSSION THREAD

**Discussion by akiroselin** [StereoKit/StereoKit#521] [Q&A]
Hi, I'm new to HoloLens and OpenXR development. Thank you for doing nice introduction about StereoKit on microsoft mixed reality dev day. I'm very attracted and wish to continue developping with stereokit. I've tried to debug using the UWP template on my HoloLens, however end up several errors. I explain below:
1. After I deploy the template example without any change and chose to run stereokitproject1 on HoloLens, it only showed a slate with SK logo on it. When I press the logo, it shows a tab saying updating/loading but then back to slate without any further progress. If I change release mode to debug mode, the app will run into error as below: "
System.TypeLoadException: 'Unresolved P/Invoke method 'kernel32!LoadLibraryW' from this method. Please look for this method in build warnings for more details.  " 
However in build warning, there are only logo link problems.... no warning related to kernel32!LoadLibraryW.
The steps I followed is from this introduction video : https://www.youtube.com/watch?v=U_7VNIcPQaM&list=RDCMUCsMica-v34Irf9KVTh6xx-g&index=1. I'm running VS2022 on Win10.

2.  When I try to run the program on laptop without deploy to hololens, but use laptop flatscreen by adding "displayPreference = DisplayMode.Flatscreen," in sk.setting, the simulation window also didnt pop out. 

I'm wondering did I anywere wrong? 
As find your tool very neat, clean and powerful, I really wish to keep develop using this tool.
Looking forward to your answers and support.
Thank you.

**Comment by maluoi** [comment]
UWP can be a bit of a fiddly platform, there's a good chance you're doing everything fine!

One thing I'd check is to make sure Visual Studio correctly resolved the right version of the StereoKit NuGet. `[Project->Manage NuGet Packages]` and make sure it's at v0.3.6. You could also "Include prerelease" packages and try out v0.3.7-preview.5, there's a couple of good fixes in there.

Another thing to try is making sure your HoloLens is up-to-date. I've had some mixed luck with the recent OS versions, but that could make a difference as well.

**Comment by akiroselin** [comment]
Dear Nick, I updated Stereokit Nuget to v0.3.7. Started new UWP template as shown in video: [StereoKit: Project Setup for HoloLens Development - YouTube]() Build was successful however still can not deploy to HoloLens. Then I try to run the project on local machine by choosing X64 platform. Got the same error as before, shown as below:
![vs_error](https://user-images.githubusercontent.com/23378063/203742135-8aece3ce-f7c3-4444-85c7-a262363b4fd7.png)
 I’m running on Visual Studio 22. Using the downloaded Stereokit UWP template from visual studio marketplace. Can I know what that error means? Or did I do somewhere wrong? Thank you for your help.

P.S nice development.

Best, Aki 

> UWP can be a bit of a fiddly platform, there's a good chance you're doing everything fine!One thing I'd check is to make sure Visual Studio correctly resolved the right version of the StereoKit NuGet. [Project->Manage NuGet Packages] and make sure it's at v0.3.6. You could also "Include prerelease" packages and try out v0.3.7-preview.5, there's a couple of good fixes in there.Another thing to try is making sure your HoloLens is up-to-date. I've had some mixed luck with the recent OS versions, but that could make a difference as well.

**Reply by maluoi** [reply]
Hey Aki, it looks like you might be replying via email, so a number of things you mentioned here got eaten when they turned into GitHub discussion posts! I edited these to make them a bit easier to read, but the links are missing, as well as any attachments.

Could you pop onto GitHub and add some of these details there?

**Reply by akiroselin** [reply]
Sorry. Updated images now.

**Comment by akiroselin** [comment]
One similar problem I find online is this:

[UWP app crash immediately after opening with Faulting module name: KERNELBASE.dll Event Log - Microsoft Q&A](https://learn.microsoft.com/en-us/answers/questions/704652/uwp-app-crash-immediately-after-opening-with-fault.html)

 

If we scroll down to bottem, it seems saying some link with SDK is missing.

If you find it useful.

And the call stack of the error is attached below:


 
![vs_callstack](https://user-images.githubusercontent.com/23378063/203741518-2d33750f-8f41-4400-ae3f-92d88024fde7.png)


Looking forward to solutions.

Sorry if I’m spamming.

Cheers,Aki

> UWP can be a bit of a fiddly platform, there's a good chance you're doing everything fine!One thing I'd check is to make sure Visual Studio correctly resolved the right version of the StereoKit NuGet. [Project->Manage NuGet Packages] and make sure it's at v0.3.6. You could also "Include prerelease" packages and try out v0.3.7-preview.5, there's a couple of good fixes in there.Another thing to try is making sure your HoloLens is up-to-date. I've had some mixed luck with the recent OS versions, but that could make a difference as well.
